client/client.go

```diff
@@ -261,22 +254,40 @@ func (c *Client) declareChannels(conn *amqp.Connection) (*amqp.Channel, *amqp.Ch
 	return inputCh, outputCh, nil
 }
 
-// runPublisher consumes messages from clientMessages and publishes them on the
+// runPublisher consumes messages from chan requests and publishes them on the
 // amqp exchange. The method will stop consuming if the underlying amqp channel
 // is closed for any reason, and when this happens the messages will be put back
-// in clientMessages unless we have retried to many times.
+// in chan requests unless we have retried to many times.
 func (c *Client) runPublisher(outChan *amqp.Channel) {
 	logger.Info("client: running publisher")
 	go func() {
-		for request := range c.clientMessages {
-			logger.Infof("client: publishing %v", request.publishing.CorrelationId)
+		for request := range c.requests {
+			replyToQueueName := ""
+			contentType := "text/plain"
```

Comment:
I don't really like the idéa of this logic here, maybe we could move it to `send()` where we populate other fields for the `Request`?

Reply:
Yes, perhaps that is better.